sanitizers/src/main/java/com/code_intelligence/jazzer/sanitizers/ScriptEngineInjection.java

```diff
@@ -0,0 +1,107 @@
+ * embedded to the JVM (they must follow the
+ * <a href="https://www.jcp.org/en/jsr/detail?id=223">JSR-223 </a>
+ * specification).
+ **/
+@SuppressWarnings("unused")
+public final class ScriptEngineInjection {
+  private static final String PAYLOAD = "\"jaz\"+\"zer\"";
+
+  /**
+   * String variants of eval can be intercepted by before hooks, as the script
+   * content can directly be checked for the presence of the payload.
+   */
+  @MethodHook(type = HookType.BEFORE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval", targetMethodDescriptor = "(Ljava/lang/String;)Ljava/lang/Object;")
+  @MethodHook(type = HookType.BEFORE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval",
+      targetMethodDescriptor = "(Ljava/lang/String;Ljavax/script/ScriptContext;)Ljava/lang/Object;")
+  @MethodHook(type = HookType.BEFORE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval",
+      targetMethodDescriptor = "(Ljava/lang/String;Ljavax/script/Bindings;)Ljava/lang/Object;")
+  public static void
+  checkScriptEngineExecuteString(
+      MethodHandle method, Object thisObject, Object[] arguments, int hookId) {
+    checkScriptContent((String) arguments[0], hookId);
+  }
+
+  /**
+   * Reader variants of eval must be intercepted by replace hooks, as their
+   * contents are converted to strings, for the payload check, and back to readers
+   * for the actual method invocation.
+   */
+  @MethodHook(type = HookType.REPLACE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval", targetMethodDescriptor = "(Ljava/io/Reader;)Ljava/lang/Object;")
+  @MethodHook(type = HookType.REPLACE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval",
+      targetMethodDescriptor = "(Ljava/io/Reader;Ljavax/script/ScriptContext;)Ljava/lang/Object;")
+  @MethodHook(type = HookType.REPLACE, targetClassName = "javax.script.ScriptEngine",
+      targetMethod = "eval",
+      targetMethodDescriptor = "(Ljava/io/Reader;Ljavax/script/Bindings;)Ljava/lang/Object;")
+  public static Object
+  checkScriptEngineExecute(MethodHandle method, Object thisObject, Object[] arguments, int hookId)
+      throws Throwable {
+    if (arguments[0] != null) {
+      String content = readAll((Reader) arguments[0]);
+      checkScriptContent(content, hookId);
+      arguments[0] = new StringReader(content);
+    }
+    return method.invokeWithArguments(thisObject, arguments);
+  }
+
+  private static void checkScriptContent(String content, int hookId) {
+    if (content != null) {
+      if (content.contains(PAYLOAD)) {
+        Jazzer.reportFindingFromHook(new FuzzerSecurityIssueCritical(
+            "Script Engine Injection: Insecure user input was used in script engine invocation.\n"
+            + "Depending on the script engine's capabilities this could lead to sandbox escape and remote code execution."));
+      }
```

Comment:
Doesn't matter except for readability, but the next line could be wrapped in an `else`.

Reply:
Done.